src/liballoc/boxed.rs

```diff
@@ -395,8 +395,7 @@ impl<T: Clone> Clone for Box<T> {
 #[stable(feature = "box_slice_clone", since = "1.3.0")]
 impl Clone for Box<str> {
     fn clone(&self) -> Self {
-        // this makes a copy of the data
-        let buf: Box<[u8]> = self.as_bytes().into();
```

Comment:
So this replaces a use of `into` by `from`. I don't see the point, but would you at least leave the comment in place please?

Reply:
> I don't see the point

I’m in agreement here; the change has a non-obvious benefit. Can you expand on why this is useful?

It also doesn’t directly make use of `Box::new`, so I’m not following the meaning of the title

Reply:
It generally seems that the code base avoids type annotations where not they aren't needed.  I think `Box::from` is clear that it is putting the bytes in a `Box` and is allocating so I don't see the need to comment this behavior.

Regarding `new`, as you can see from the commit log, I originally replaced it with `new` but that didn't work because it made a `Box<&[u8]>`.

Reply:
>  I think Box::from is clear that it is putting the bytes in a Box and is allocating so I don't see the need to comment this behavior.

`Box::from` where the argument is a `Box` or `Vec` does not allocate. So I don't think this is clear at all.

Reply:
I'm just going to close I don't think it really matters that much.